### ISSUE TITLE
Use threads when writing code

### ISSUE THREAD

**Issue by KeyWorksRW** [KeyWorksRW/wxUiEditor#877]
### Description:

Switching generator functions to use the new `Code` class for generating C++/Python code definitely makes the code in the generator functions easier to maintain. However, it's also a bit slower to generate given the need for branching within the class to handle variations in C++/Python code output, and of course the new line breaking algorithm. Once editable files are generated (see #875) code generation will get even slower given that we will now need to parse the file we read from disk. I've noticed that the internal Code Comparison has gotten noticeably slower when run wxUiEditor's own project with it's 40+ forms.

It needs a thorough code review, but I'm fairly certain that the individual code writing functions simply return warning/error results without using any UI. That means that it _should_ be possible to handle all code writing via threads.

On the technical side, using the ttLib class for thread pooling would be ideal since it's very efficient at knowing exactly which thread is available for more work. Unfortunately, only Windows supports an efficient method of checking which thread is available, and while we currently require Wine to run on Unix/Mac, that's only a build issue. Eventually, I want wxUiEditor to at least run natively on Unix. That means we have to use a slower, inefficient method to check for thread availability so that someday we can run on Unix.

This does _not_ apply to the code panels -- those are always working with a single form, so they are already generating at maximum speed.

Note that issue #670 (enable command-line code generation) will also rely on there being no UI messages while writing code. It too will need a thorough code review to ensure the lack of UI while writing code.

**Comment by Randalphwa** [comment]
I created a single function that could either be called from a created thread, or from the main code of execution. In Debug/Internal builds, I added a timer and reported on the results allowing for comparison between normal and threaded processing. The function is passed a form and does all code generation, comparing the result with any file on disk, and writing the file if it has changed.

I used a thread pool consisting of the number of CPUs / 3 (if nCPUS > 3). Increasing that number slowed things down, probably due to thread creation overhead, and the fact that the called function will briefly create 3 threads of it's own.

I tested the result on a project with 19 forms, using a CPU capable of 24 concurrent threads. What I found is that the threaded version was slightly _slower_ then the non-thread version. I also tested just code comparison (it does everything except write the updated files) on the wxUiEditor project itself with it's current 39 forms. In this case, using threads reduced the amount of time by about 30 milliseconds -- which obviously is not enough to care about.

I tried switching from reading using `std::ifstream` to `stdio::_read` which made no difference (entire file read at once either way). I tried increasing the initial buffer size in `FileCodeWriter` which also made no difference. I did verify that the three threads created in `GenerateBaseClass()` do improve performance by a small fraction.

Note to self -- while it is a high-end machine, generating code for 39 files in a third of a second using a Debug build makes me think I should stop worrying about performance...

tldr; -- using a thread pool for the outer code generation routine hurts, rather than improves performance.
